Tests/MapboxNavigationTests/RouteDurationTests.swift

```diff
@@ -0,0 +1,91 @@
+import XCTest
+import TestHelper
+import Turf
+import MapboxMaps
+@testable import MapboxNavigation
+
+class RouteDurationTests: TestCase {
+    var navigationMapView:NavigationMapView = NavigationMapView(frame: CGRect(origin: .zero, size: .iPhone6Plus))
+    enum RouteDurationAnnotationTailPosition: Int {
+        case left
+        case right
+    }
+    let routeDurationAnnotationsSource = NavigationMapView.SourceIdentifier.routeDurationAnnotationsSource
+    let routeDurationAnnotationsLayer = NavigationMapView.LayerIdentifier.routeDurationAnnotationsLayer
+
+    override func setUp() {
+        super.setUp()
+    }
+    
+    override func tearDown() {
+        navigationMapView.removeRouteDurations()
+        super.tearDown()
+    }
+    
+    func setupRouteDurationsLayer(tailPosition: RouteDurationAnnotationTailPosition) throws {
+        let style = navigationMapView.mapView.mapboxMap.style
+        
+        let coordinate: CLLocationCoordinate2D = CLLocationCoordinate2D(latitude: 37.79129, longitude: -122.396152)
+        var feature = Feature(geometry: .point(Point(coordinate)))
+        feature.properties = ["tailPosition": tailPosition.rawValue, "text": "testLabel"]
+        
+        var dataSource = GeoJSONSource()
+        dataSource.data = .feature(feature)
+        try style.addSource(dataSource, id: routeDurationAnnotationsSource)
+        
+        var shapeLayer = SymbolLayer(id: routeDurationAnnotationsLayer)
+        shapeLayer.iconTextFit = .constant(IconTextFit.both)
+        shapeLayer.iconAllowOverlap = .constant(true)
+        shapeLayer.textAllowOverlap = .constant(true)
+        shapeLayer.textJustify = .constant(TextJustify.left)
+        shapeLayer.symbolZOrder = .constant(SymbolZOrder.auto)
+        shapeLayer.source = routeDurationAnnotationsSource
+        shapeLayer.textSize = .constant(16)
+        shapeLayer.textField = .expression(Exp(.get) {
+            "text"
+        })
+        
+        let anchorExpression = Exp(.match) {
+            Exp(.get) { "tailPosition" }
+            0
+            "bottom-left"
+            1
+            "bottom-right"
+            "center"
+        }
+        shapeLayer.textAnchor = .expression(anchorExpression)
+        
+        let offsetExpression = Exp(.match) {
+            Exp(.get) { "tailPosition" }
+            0
+            Exp(.literal) { [0.5, -1.0] }
+            Exp(.literal) { [-0.5, -1.0] }
+        }
+        shapeLayer.textOffset = .expression(offsetExpression)
+        
+        try style.addLayer(shapeLayer)
+    }
+    
+    func testTailPositionLeft() throws {
```

Comment:
Test should also involve verification that any change to the `Feature` property, which is used in source will affect layer outlook. E.g. change to `tailPosition` will change its `textAnchor` or `textOffset`. 

Think of test as an entity, which simplifies your life and does the job you'd do manually (e.g. to verify that certain feature works as expected), but a lot faster.

Reply:
Right now it seems unable to test the `Expression` from our side or map side as in the [ticket](https://github.com/mapbox/mapbox-maps-ios/issues/556).